src/asset-swapper/quote_consumers/feature_rules/abstract_feature_rule.ts

```diff
@@ -0,0 +1,30 @@
+import { BigNumber } from '@0x/utils';
+import { SwapQuote, ExchangeProxyContractOpts, CalldataInfo } from '../../types';
+import { getMaxQuoteSlippageRate } from '../quote_consumer_utils';
+import { FeatureRule, SwapContext } from './types';
+
+export abstract class AbstractFeatureRule implements FeatureRule {
+    /** Returns a commonly needed context.*/
+    protected getSwapContext(quote: SwapQuote, opts: ExchangeProxyContractOpts): SwapContext {
+        // Take the bounds from the worst case
+        const sellAmount = BigNumber.max(
+            quote.bestCaseQuoteInfo.totalTakerAmount,
+            quote.worstCaseQuoteInfo.totalTakerAmount,
+        );
+
+        const ethAmount = quote.worstCaseQuoteInfo.protocolFeeInWeiAmount.plus(opts.isFromETH ? 0 : sellAmount);
+        const maxSlippage = getMaxQuoteSlippageRate(quote);
+
+        return {
+            sellToken: quote.takerToken,
```

Comment:
for my understanding -- is this always the case in general or just for VIP features?